flink-clients/src/test/java/org/apache/flink/client/deployment/application/ClassPathPackagedProgramRetrieverTest.java

```diff
@@ -312,87 +205,64 @@ public void testRetrieveCorrectUserClasspathsWithSpecifiedEntryClass()
 
         assertThat(
                 jobGraph.getClasspaths().stream().map(URL::toString).collect(Collectors.toList()),
-                containsInAnyOrder(expectedURLs.stream().map(URL::toString).toArray()));
+                containsInAnyOrder(EXPECTED_URLS.stream().map(URL::toString).toArray()));
     }
 
     @Test
-    public void testRetrieveFromJarFileWithoutUserLib()
-            throws IOException, FlinkException, ProgramInvocationException {
-        final File testJar = TestJob.getTestJobJar();
-        final ClassPathPackagedProgramRetriever retrieverUnderTest =
-                ClassPathPackagedProgramRetriever.newBuilder(PROGRAM_ARGUMENTS)
-                        .setJarFile(testJar)
-                        .build();
-        final JobGraph jobGraph = retrieveJobGraph(retrieverUnderTest, new Configuration());
+    public void testGetPackagedProgramWithConfiguration() throws IOException, FlinkException {
```

Comment:
This should also be tested for JarFilePackagedProgramRetriever and PythonBasedPackagedProgramRetriever.